src/main/java/com/devcycle/sdk/server/local/bucketing/LocalBucketing.java

```diff
@@ -203,7 +203,7 @@ private byte[] readAssemblyScriptUint8Array(int address)
         return bufferData;
     }
 
-    public void storeConfig(String sdkKey, String config) {
+    public synchronized void storeConfig(String sdkKey, String config) {
```

Comment:
This only protects against multiple calls to the same function - instead of protecting against concurrent calls into WASM itself. 
I think a mutex is the better way around this

Reply:
Eg - if this was to be instantiated more than once - each instance of the class would be protected - but not against eachother.

Reply:
or - the parent WASM init functions: 
https://github.com/DevCycleHQ/java-server-sdk/blob/b1a26102dc8ccfb1b9f99206ecf251217dbc6515/src/main/java/com/devcycle/sdk/server/local/bucketing/LocalBucketing.java#L71

Reply:
so in Java when you use the synchronized keyword, it just uses the intrinsic mutex in the object instance, much like doing:
```
public void someMethod() {
  synchronized(this) {
    // my thread safe stuff
  }
}
```

it is just using the instance as the mutex, so all synchronized methods in the same instance are thread safe with each other. That said, i could just rework it to use an explicit mutex object instance and synchronize on that so it matches the other SDKS and doesn't rely on java-isms

Reply:
Interesting - I'm happy to keep it with the "Java" version - as that'd probably be safer than implementing a mutex manually.